x/concentrated-liquidity/internal/math/math.go

```diff
@@ -12,9 +14,15 @@ func Liquidity0(amount sdk.Int, sqrtPriceA, sqrtPriceB sdk.Dec) sdk.Dec {
 	if sqrtPriceA.GT(sqrtPriceB) {
 		sqrtPriceA, sqrtPriceB = sqrtPriceB, sqrtPriceA
 	}
-	product := sqrtPriceA.Mul(sqrtPriceB)
-	diff := sqrtPriceB.Sub(sqrtPriceA)
-	return amount.ToDec().Mul(product).Quo(diff)
+
+	// transform sqrtPriceA and sqrtPriceB to BigDec using osmomath for more final precision
```

Comment:
Is precision loss significant enough to require this?

I'm surprised that basic math operations are causing a large precision loss

Reply:
I think using `osmomath.BigDec` is useful in `TickToSqrtPrice` since it uses functions such as `Power`. However, I would be surprised if multiplications, subtractions and additions would be the cause of precision loss here

Reply:
![Screenshot 2022-12-19 at 10 34 01 AM](https://user-images.githubusercontent.com/40078083/208474441-00573efc-1e53-4c25-9c95-b59d9641e4f9.png)

Reverting this back to what it was produces very minimal significance, I was just trying to match the exact result of wolframalpha. Do you think I should revert this to what is was (no using osmomath) and just accept the extremely small precision loss?

Reply:
Offline, agreed to revert liquidity0 and liquidity1 back to normal sdk.Decs due to minimal improvement in precision

Reply:
After further tests, the precision loss is actually significant enough to warrant utilize BigDecs here. I will add them back and we can discuss further